src/apscheduler/datastores/mongodb.py

```diff
@@ -354,15 +354,17 @@ async def get_next_schedule_run_time(self) -> datetime | None:
         async for attempt in self._retry():
             with attempt:
                 document = self._schedules.find_one(
-                    {"next_run_time": {"$ne": None}},
-                    projection=["next_run_time"],
-                    sort=[("next_run_time", ASCENDING)],
+                    {"next_fire_time": {"$ne": None}},
+                    projection=["next_fire_time"],
+                    sort=[("next_fire_time", ASCENDING)],
                 )
 
-        if document:
-            return document["next_run_time"]
-        else:
-            return None
+            if document:
+                return document["next_fire_time"]
+            else:
+                return None
```

Comment:
Why did you indent this code here?

Reply:
In the original, document is potentially undefined depending on how the replays play out. I'll revert it since this is probably better addressed separately.

Reply:
If retries are exhausted, an exception should be raised, so the document would always be defined, yes?

Reply:
Ah, I see, makes sense.